solution/ui/prototype/src/views/Part.vue

```diff
@@ -147,31 +276,96 @@ export default {
         tabsContent() {
             return [this.tocContent, this.partContent, null, null];
         },
-        contentContainerResourcesClass() {
-            return `content-container-${this.resourcesDisplay}`;
-        },
     },
 
     async created() {
-        this.$watch(
-            () => this.$route.params,
-            (toParams, previousParams) => {
-                // react to route changes...
-                if (toParams.part !== previousParams.part) {
-                    this.structure = null;
-                    this.clearResourcesParams();
-                    this.title = toParams.title;
-                    this.part = toParams.part;
-                    this.resourcesDisplay =
-                        toParams.resourcesDisplay || "drawer";
-                }
+        for (const key of Object.keys(this.queryParams)) {
```

Comment:
Why a for loop here?  This seems to be looking for 2 specific keys and then doing the same thing if it is.  

`if (this.queryParams.subpart){ this.tabLabels["subpart"] = this.formatTabLabel("subpart")
else if (this.queryParams.section){ this.tabLabels["section"] = this.formatTabLabel("section"}`